### ISSUE TITLE
Override Keyword Specified On Class Functions When Using Base Class Switch

### ISSUE THREAD

**Issue by perlmunger** [rentzsch/mogenerator#253]
When I pass in the --base-class switch to the command line arguments, the machine generated file uses override for the entityName() and entity(ManagedObjectContext) methods. Like so:

```
class _AssetMO: BaseManagedObject {
    // MARK: - Class methods

    override class func entityName () -> String {
        return "Asset"
    }

    override class func entity(managedObjectContext: NSManagedObjectContext!) -> NSEntityDescription! {
        return NSEntityDescription.entityForName(self.entityName(), inManagedObjectContext: managedObjectContext);
    }
...
}
```
This same model when generated without the base class, does not emit the override keyword.
```
class _AssetMO: NSManagedObject {

    // MARK: - Class methods

    class func entityName () -> String {
        return "Asset"
    }

    class func entity(managedObjectContext: NSManagedObjectContext!) -> NSEntityDescription! {
        return NSEntityDescription.entityForName(self.entityName(), inManagedObjectContext: managedObjectContext);
    }

...
}
```

The override causes a compiler error:
"Method does not override any method from its superclass"

My exact mogenerator command is:

```
/usr/local/bin/mogenerator --swift --base-class=BaseManagedObject -m ShareStuff.xcdatamodeld/ShareStuff.xcdatamodel  --output-dir ./ModelObjects
```

**Comment by dmdeller** [comment]
I noticed this too. It seems like it may be an intentional feature: if you have these methods defined in your base class, then you can call them on an object declared as your base class (rather than a specific subclass). Otherwise, you wouldn't be able to, since these methods are not part of NSManagedObject.

It's probably obvious, but in case it helps someone else who has this problem - putting these method definitions in the base class worked for me:

    class func entityName () -> String
    {
        return NSStringFromClass(self)
    }
    
    class func entity(managedObjectContext: NSManagedObjectContext!) -> NSEntityDescription!
    {
        return NSEntityDescription.entityForName(self.entityName(), inManagedObjectContext: managedObjectContext);
    }

Of course, it doesn't really matter what the methods in your base class do, since they will be overridden by the mogenerator ones.

A better solution would be to put abstract method definitions (without corresponding implementations) in the base class, but I'm not sure if Swift supports this (I couldn't find mention of it).

**Comment by perlmunger** [comment]
I ended up doing what @dmdeller suggested. That seems to be the most straight forward way for my project and it wasn't a big deal since I was already using a base class for other common functionality.

**Comment by atomicbird** [comment]
I'm reopening this because including this code in the base class seems like an ugly solution. The base class doesn't represent an entity, so it shouldn't need code like `entityName` that will just be overridden. I'm working on a fix that won't require that, but should correctly add `override` when the entity and class hierarchy require it. More detail is in the discussion of pull request #257

**Comment by perlmunger** [comment]
Sounds great @atomicbird

**Comment by atomicbird** [comment]
Fixed in 78e62b1 as described in issue #257
